### PR TITLE
[bugfix] fix slow accounts / statuses using emojis lookups

### LINKED ISSUE #2015: [bug/investigation] Slow DB query selecting statuses using specific emoji
From an email:

> You can run the following query by
> hand and see a HUGE delay:
> 
> timestamp="22/07/2023 01:01:04.170" func=bundb.queryHook.AfterQuery
> level=WARN duration=16.497780301s query="SELECT \"id\" FROM \"statuses\"
> WHERE ('01WK9MJ69CGFHT4D9H3RAZKHKY' IN (emojis))" msg="SLOW DATABASE QUERY"
> 
> (My instance has about a million rows in the statuses table.)
> 
> This still happens on rc3.  This abort was after 16.5 seconds, but it
> actually takes > 1 minute (maybe more than the DB timeout config value).
> The service grinds to a halt, and my hosting provider emails me about
> pegged CPU.

### PR DESCRIPTION
**Pull request by NyaaaWhatsUpDoc**
## Description

Updates emoji delete, and get statuses / accounts using emoji functions to use their appropriate relational tables for faster lookups.

closes #2015 

## Checklist
- [x] I/we have read the [GoToSocial contribution guidelines](https://github.com/superseriousbusiness/gotosocial/blob/main/CONTRIBUTING.md).
- [x] I/we have discussed the proposed changes already, either in an issue on the repository, or in the Matrix chat.
- [x] I/we have performed a self-review of added code.
- [x] I/we have written code that is legible and maintainable by others.
- [x] I/we have commented the added code, particularly in hard-to-understand areas.
- [ ] I/we have made any necessary changes to documentation.
- [ ] I/we have added tests that cover new code.
- [x] I/we have run tests and they pass locally with the changes.
- [x] I/we have run `go fmt ./...` and `golangci-lint run`.

### CODE REVIEW

File: internal/db/bundb/emoji.go [hunk 1]
```diff
@@ -106,47 +106,36 @@ func (e *emojiDB) DeleteEmojiByID(ctx context.Context, id string) error {
...
-
-		// Select all accounts using this emoji into accountIDss.
-		if _, err := aq.Exec(ctx, &accountIDs); err != nil {
+		// Delete relational links between this emoji
+		// and any accounts using it, returning the
+		// account IDs so we can later update them.
+		if _, err := tx.NewDelete().
+			Table("account_to_emojis").
+			Where("? = ?", bun.Ident("emoji_id"), id).
+			Returning("account_id").
```
Comment (tsmethurst) [inline internal/db/bundb/emoji.go:126]: Clever!

File: internal/db/bundb/status.go [hunk 2]
```diff
@@ -43,7 +43,6 @@ func (s *statusDB) newStatusQ(status interface{}) *bun.SelectQuery {
 	return s.db.
 		NewSelect().
 		Model(status).
-		Relation("Tags").
```
Comment (NyaaaWhatsUpDoc) [inline internal/db/bundb/status.go:46]: we're already doing this in PopulateStatus() now

File: internal/db/bundb/emoji.go (hunk 1 above)
Comment (NyaaaWhatsUpDoc) [reply internal/db/bundb/emoji.go:126]: :D

File: internal/db/bundb/account.go [hunk 3]
```diff
@@ -468,24 +468,13 @@ func (a *accountDB) GetAccountCustomCSSByUsername(ctx context.Context, username
...
-	// text anyways, allowing a simple LIKE query.
-	q = whereLike(q, "emojis", emojiID)
-
-	// Execute the query, scanning destination into accountIDs.
-	if _, err := q.Exec(ctx, &accountIDs); err != nil {
+	// SELECT all accounts using this emoji,
+	// using a relational table for improved perf.
+	if _, err := a.db.NewSelect().
+		Table("account_to_emojis").
+		Column("id").
```
Comment (NyaaaWhatsUpDoc) [inline internal/db/bundb/account.go:475]: ah shit need to fix this :grimacing:

Comment (NyaaaWhatsUpDoc) [reply internal/db/bundb/account.go]: and done

**Pull request merged**